Textbox

vincetiu8 opened the pull request:
Adds a textbox into the game, which can be used to insert dialogue or give tips to the players. Text can be added to a level by editing the level.json (I don't feel like adding support to dev tools for this anymore) as a sub wave:
```
...
"waves: {
[
"Example text here."
"More text here."
]
...
```

An example of this can be seen in level 1.

vincetiu8 committed:
Support inserting texts directly into level json

vincetiu8 committed:
Add a textbox

vincetiu8 committed:
Fix bugs

Graphcalibur reviewed (changes requested):
A strange bug occurs if you restart the level after scrolling through the text and hitting "Start Wave". Instead of the level going through the text again, the level will immediately begin sending enemies out even if you don't press the "Start Wave" button.

vincetiu8 committed:
Fix start bug

This fixes a bug where if the game was restarted it would spawn enemies and ignore the text.

Graphcalibur reviewed (approved):
LGTM!
